### DISCUSSION TITLE
The best approach to use 2nd disk for local volume/hostpath data

### DISCUSSION THREAD

**Discussion by d-lavayen** [okd-project/okd#2039] [General]
For about a year now, I've been using the local volume/hostpath type storage for my databases. Recently, I’ve started to question whether my approach to managing data is correct from the perspective of OKD operations. My workflow involves creating a special MachineConfig that, at every node startup, mounts a second disk to the /var/csi path, where container data is stored.

I’m not entirely sure if this is the right approach, and I’d like to ask if anyone has developed a better method.

I use two disks because, in the event that the disk fills up, only the disk and applications using it will be affected, rather than the entire worker node along with the cluster containers.

I’ve considered Rook Ceph, but from what I’ve read, replication isn’t the best option for databases. Moreover, it seems to consume a lot of resources.

I’d be happy to hear how you manage data in your environments.

**Comment by Goose29** [comment]
From personal experience, I have had no issues running databases on top of Rook Ceph for medium sized workloads.

Yes, it consumes a lot of resources. IF you don't apply resource limits, then it will just take what is available. Put sensible limits on, and then it generally behaves.

**Comment by akashsemil** [comment]
I won't recommend running workloads using hostPath, What happens if that nodes goes down ? 

Ceph is best, I have been running it for years, If you are looking something with low resources try longhorn
